presto-orc/src/main/java/io/prestosql/orc/OrcWriter.java

```diff
@@ -445,9 +445,9 @@ public void close()
         stats.updateSizeInBytes(-previouslyRecordedSizeInBytes);
         previouslyRecordedSizeInBytes = 0;
 
-        flushStripe(CLOSED);
-
-        orcDataSink.close();
+        try (Closeable ignored = orcDataSink) {
+            flushStripe(CLOSED);
+        }
```

Comment:
Can this lead to situation similar to https://github.com/prestosql/presto/issues/3326 ?

Reply:
It's possible. Do you know what data source this occurs for?